src/GraphClient.ts

```diff
@@ -50,25 +55,68 @@ export default class GraphClient {
     // GraphQL query that is parsed into the standard GraphQL AST(Abstract syntax tree)
     const gqlSubscriptionQry = gql`${subscriptionQry}`;
     // Subscription handling
-    const querySubscriber = await Promise.resolve(this.apolloClient.subscribe({
+    const observable = this.apolloClient.subscribe({
       query: gqlSubscriptionQry,
       variables: {},
-    }).subscribe({
-      async next(response) {
-        const transactions = await fetcher.fetch(response.data);
-        await handler.handle(transactions);
-        // Integrate updation of ContractEntity uts here. Handlers should make sure error
-        // Promise.reject is thrown on error cases.
-      },
-      error(err) {
-        // Log error using logger
-        Logger.error(err);
-      },
-    }));
+    });
+    const querySubscriber = await Promise.resolve(
+      observable
+        .subscribe({
+          async next(response) {
```

Comment:
Can we add types to `response`, `transactions` to ease readability?

Reply:
Done